src/typing.py

```diff
@@ -1218,24 +1216,20 @@ def __getitem__(self, parameters):
         args, result = parameters
         if args is ...:
             parameters = (..., result)
-        elif args == []:
-            parameters = ((), result)
         else:
             if not isinstance(args, list):
                 raise TypeError("Callable[args, result]: args must be a list."
                                 " Got %.100r." % (args,))
-            parameters = tuple(args) + (result,)
+            parameters = tuple(args), result
         return self.__getitem_inner__(parameters)
 
     @_tp_cache
     def __getitem_inner__(self, parameters):
-        *args, result = parameters
+        args, result = parameters
         msg = "Callable[args, result]: result must be a type."
         result = _type_check(result, msg)
-        if args == [...,]:
+        if args == ...:
```

Comment:
I'd spell this as `Ellipsis` -- seeing `args == ...` is just too weird. :-) Plus it's a singleton, so you should use `is`.